### PR TITLE
Pu/isformula

### LINKED ISSUE #781: Array formulas should either be classed as formula & array type or not.
### Description
There's a mismatch in functions for the cells with an ARRAY type that are also formulas.

See here:

image

`doesCellHaveFormula()` returns false.
But `getSheetFormulas()` returns it as part of it's results.

This is mis-leading to the user. It should either be classes as a formula or not but not in between.

### Steps to reproduce
```js
const sheetName = hf.addSheet(state.currentSheet);
const sheetId = hf.getSheetId(sheetName);

// Fill the HyperFormula sheet with data.
hf.setSheetContent(sheetId, [[0, 1]]);
hf.setCellContents(
  { sheet: sheetId, row: 0, col: 2 },
  "=ARRAYFORMULA(ISEVEN(A1:A2*10))"
);

console.log(hf.doesCellHaveFormula({ sheet: sheetId, row: 0, col: 2 }));
console.log(hf.getSheetFormulas(sheetId));
```

### Demo
https://codesandbox.io/s/fervent-cannon-voj87?file=/src/hyperformulaConfig.js

### Your environment

* HyperFormula version: 1.0.0
* Browser Name and version: Chrome latest
* Operating System: Windows 10

### PR DESCRIPTION
**Pull request by izulin**
### Context

Fixes #781 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

File: src/Cell.ts [hunk 1]
```diff
@@ -54,18 +55,23 @@ export enum CellType {
   VALUE = 'VALUE',
   ARRAY = 'ARRAY',
   EMPTY = 'EMPTY',
+  ARRAYFORMULA = 'ARRAYFORMULA',
```
Comment (wojciechczerniak) [inline src/Cell.ts:58]: Pls add it on docs page as well: https://github.com/handsontable/hyperformula/blob/master/docs/guide/types-of-values.md

File: test/engine.spec.ts [hunk 2]
```diff
@@ -556,7 +556,7 @@ describe('#getCellType', () => {
   it('formula matrix', () => {
     const engine = HyperFormula.buildFromArray([['=TRANSPOSE(C1:C2)']])
 
-    expect(engine.getCellType(adr('A1'))).toBe(CellType.ARRAY)
+    expect(engine.getCellType(adr('A1'))).toBe(CellType.ARRAYFORMULA)
     expect(engine.getCellType(adr('B1'))).toBe(CellType.ARRAY)
   })
```
Comment (wojciechczerniak) [inline test/engine.spec.ts:561]: Can we add test with an array that is not a result of a formula? With `{1,2,3;4,5,6}` we will have `CellType.ARRAY` returned for all cells, A1 included, right?

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: test/engine.spec.ts (hunk 2 above)
Comment (izulin) [reply test/engine.spec.ts:561]: I don't think it is possible to create array without formula.

Comment (wojciechczerniak) [reply test/engine.spec.ts:561]: Ok, I forgot that `=` is needed

https://github.com/handsontable/hyperformula/blob/413e199a632452bffb45f33c747b53a6f8af2cab/test/arrays.spec.ts#L234-L237

**Review by wojciechczerniak — APPROVED** [review]
👍

**Pull request merged**